src/yz_kv.erl

```diff
@@ -578,30 +580,49 @@ siblings_permitted_test_() ->
                       Object = riak_object:new(B, K, V),
                       CheckBucket = riak_object:bucket(Object),
                       CheckBucketProps = riak_core_bucket:get_bucket(CheckBucket),
-                      ?assertNot(siblings_permitted(CheckBucketProps))
+                      ?assertNot(siblings_permitted(Object, CheckBucketProps))
                   end || {B, K, V} <- [{Bucket1, <<"k1">>, hi},
                                      {Bucket2, <<"k2">>, hey}]]
              end),
       ?_test(begin
                  BucketType1 = <<"counters">>,
                  BucketType2 = <<"maps">>,
+                 BucketType3 = <<"loopfors">>,
                  Bucket1 = {BucketType1, <<"crdt">>},
                  Bucket2 = {BucketType2, <<"crdtz">>},
-                 riak_core_bucket_type:create(BucketType1, [{datatype, counter}]),
+                 Bucket3 = {BucketType3, <<"crdtbz">>},
+                 riak_core_bucket_type:create(BucketType1,
+                                              [{datatype, counter}]),
                  riak_core_bucket_type:activate(BucketType1),
                  riak_core_bucket_type:create(BucketType2, [{datatype, map}]),
                  riak_core_bucket_type:activate(BucketType2),
+                 riak_core_bucket_type:create(BucketType3, []),
+                 riak_core_bucket_type:activate(BucketType3),
                  BTProps1 = riak_core_bucket:get_bucket(Bucket1),
                  BTProps2 = riak_core_bucket:get_bucket(Bucket2),
+                 BTProps3 = riak_core_bucket:get_bucket(Bucket3),
                  ?assertEqual(counter, proplists:get_value(datatype, BTProps1)),
                  ?assertEqual(map, proplists:get_value(datatype, BTProps2)),
+                 ?assertEqual(undefined, proplists:get_value(datatype,
+                                                             BTProps3)),
                  [begin
                       Object = riak_object:new(B, K, V),
                       CheckBucket = riak_object:bucket(Object),
                       CheckBucketProps = riak_core_bucket:get_bucket(CheckBucket),
-                      ?assertNot(siblings_permitted(CheckBucketProps))
+                      ?assert(siblings_permitted(Object, CheckBucketProps))
```

Comment:
I don't quite understand why the logic is reversed in this check.  The objects in question are in a counter and map bucket, respectively.  Does the is_datatype check (i.e., riak_kv_crdt:is_crdt/2) return a different value in the most recent riak_kv version?